tests/query/processors/test_match_replace_processor.py

```diff
@@ -0,0 +1,102 @@
+import pytest
+
+from snuba.clickhouse.columns import ColumnSet
+from snuba.clickhouse.formatter import ClickhouseExpressionFormatter
+from snuba.datasets.schemas.tables import TableSource
+from snuba.query.expressions import Column, Expression, FunctionCall, Literal
+from snuba.query.logical import Query, SelectedExpression
+from snuba.query.matchers import (
+    Column as ColumnMatch,
+    String as StringMatch,
+    MatchResult,
+    Param,
+)
+from snuba.query.processors.match_replace_processor import MatchReplaceProcessor
+from snuba.request.request_settings import HTTPRequestSettings
+
+test_data = [
+    (
```

Comment:
Please use pytest.param and assign an id to each case so that they can be identified easily when they fail.